simulation-runner/src/main/scala/com/puppetlabs/gatling/node_simulations/PerfTestLarge.scala

```diff
@@ -0,0 +1,72 @@
+package com.puppetlabs.gatling.node_simulations
+import com.puppetlabs.gatling.runner.SimulationWithScenario
+import org.joda.time.LocalDateTime
+import org.joda.time.format.ISODateTimeFormat
+import java.util.UUID
+
+import scala.concurrent.duration._
+
+import io.gatling.core.Predef._
+import io.gatling.http.Predef._
+
+class PerfTestLarge extends SimulationWithScenario {
+
+	val reportBody = ElFileBody("PerfTestLarge_0006_request.txt")
+
+	val baseHeaders = Map("Accept" -> "application/json, text/pson",
+		"Accept-Encoding" -> "gzip;q=1.0,deflate;q=0.6,identity;q=0.3",
+		"User-Agent" -> "Puppet/5.3.3 Ruby/2.4.2-p198 (x86_64-linux)")
+
+	val headers_0 = baseHeaders ++ Map("X-Puppet-Version" -> "5.3.3")
+
+	val headers_6 = baseHeaders ++ Map(
+		"Connection" -> "close",
+		"Content-Type" -> "application/json",
+		"X-Puppet-Version" -> "5.3.3")
+
+	val scn = scenario("samtest")
+		.exec(http("node")
+			.get("/puppet/v3/node/${node}?environment=production&transaction_uuid=feaf75a3-c82c-4e06-9d02-5387f14034a0&fail_on_404=true")
+			.headers(headers_0))
+		.pause(103 milliseconds)
+		.exec(http("filemeta pluginfacts")
+			.get("/puppet/v3/file_metadatas/pluginfacts?environment=production&links=follow&recurse=true&source_permissions=use&ignore=.svn&ignore=CVS&ignore=.git&ignore=.hg&ignore=%2A.pot&checksum_type=md5")
+			.headers(headers_0))
+		.pause(1)
+		.exec(http("filemeta plugins")
+			.get("/puppet/v3/file_metadatas/plugins?environment=production&links=follow&recurse=true&source_permissions=ignore&ignore=.svn&ignore=CVS&ignore=.git&ignore=.hg&ignore=%2A.pot&checksum_type=md5")
+			.headers(headers_0))
+		.pause(1)
+		.exec(http("")
+			.get("/puppet/v3/file_metadatas/locales?environment=production&links=follow&recurse=true&source_permissions=ignore&ignore=.svn&ignore=CVS&ignore=.git&ignore=.hg&ignore=%2A.pot&ignore=config.yaml&checksum_type=md5")
```

Comment:
so... this will only work in Irving+, IIRC
i'm probably okay with that.

Reply:
oh.  the `X-Puppet-Version` header will probably limit how far we can take this "agent" as well, yeah?

Reply:
Yes and yes. We will probably need to modify this file slightly per version/release. We still need to figure out what the best way to catch changes to the agent runs should be so that we can ensure the requests stay up to date.  Both sound like good future tickets.

Reply:
i don't think we know the abstraction yet.  i'd hope we'd program it in as parameters to a function in the gatling stuff, or some similar abstraction

Reply:
No, what it means is that this file will have some variables in it that need to be replaced depending on which version we are running.  This works for current version, but some work will need to be done to make it work for all versions going forward.  We may be able to use some of Gatling's tooling or scala code to include or exclude different requests based on the build we are testing, or we may need to do something dirty like parse and modify this file...

Reply:
sounds like a good candidate to ticket for later consideration.

Reply:
Created QA-3445